app/models/fetcher_base.rb

```diff
@@ -1,6 +1,11 @@
 class FetcherBase
   STRATEGIES = [ :fetch_all, :fetch_minimal ]
 
+  def self.for(reducers)
```

Comment:
Minor: still not a fan of this name and the way this is used, it would be clearer to me when using the subclass to set the strategy through a constructor / setter and then run the fetch (i.e. over 2 lines). 

Alternatively the method name could be `build_with_fetch_strategy` ?

Reply:
Removed this completely and just moved this logic into the constructor.